R/tag_query.R

```diff
@@ -306,7 +303,31 @@ tagEnvToTags_ <- function(x) {
     # undo parent env and key
     x$parent <- NULL
     x$envKey <- NULL
-    # recurse through children
+    xNames <- names(x)
+
+    # Reorder the names to match a typical tag() structure that has `name`, `attribs`, and `children` first
+    # Avoid calling `setdiff()` if possible (it is slow).
+    if (!is.character(x[["name"]])) {
+      stop("A tag environment has lost its `$name`. Did you remove it?")
+    }
+    if (is.null(x[["attribs"]])) x$attribs <- setNames(list(), character(0))
+    if (is.null(x[["children"]])) x$children <- list()
+
+    newNames <- c(
+      "name", "attribs", "children",
+      if (
+        length(xNames) > 3
```

Comment:
Should you be computing `xNames` _after_ filling in defaults?

Reply:
Yes!